ledger/store/trackerdb/generickv/accounts_ext_reader.go

```diff
@@ -0,0 +1,521 @@
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/ledger/store/trackerdb"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+func (r *accountsReader) AccountsRound() (rnd basics.Round, err error) {
+	// SQL at time of impl:
+	//
+	// "SELECT rnd FROM acctrounds WHERE id='acctbase'"
+
+	// read round entry
+	value, closer, err := r.kvr.Get(roundKey())
+	if err != nil {
+		return
+	}
+	defer closer.Close()
+
+	// parse the bytes into a u64
+	rnd = basics.Round(binary.BigEndian.Uint64(value))
+
+	return
+}
+
+func (r *accountsReader) AccountsTotals(ctx context.Context, catchpointStaging bool) (totals ledgercore.AccountTotals, err error) {
+	// read round entry
+	value, closer, err := r.kvr.Get(totalsKey(catchpointStaging))
+	if err != nil {
+		return
+	}
+	defer closer.Close()
+
+	err = protocol.Decode(value, &totals)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (r *accountsReader) AccountsHashRound(ctx context.Context) (hashrnd basics.Round, err error) {
+	// TODO: catchpoint
+	return
+}
+
+func (r *accountsReader) LookupAccountAddressFromAddressID(ctx context.Context, ref trackerdb.AccountRef) (address basics.Address, err error) {
+	// TODO: catchpoint
+	return
+}
+
+func (r *accountsReader) LookupAccountRowID(addr basics.Address) (ref trackerdb.AccountRef, err error) {
+	// Note: [perf] this is not a very cheap operation since we have to pull up the entire record
+	acc, err := r.LookupAccount(addr)
+	if err != nil {
+		return
+	}
+
+	if acc.Ref == nil {
+		return nil, trackerdb.ErrNotFound
+	}
+
+	return acc.Ref, nil
+}
+
+func (r *accountsReader) LookupResourceDataByAddrID(accRef trackerdb.AccountRef, aidx basics.CreatableIndex) (data []byte, err error) {
+	// TODO: this can probably get removed in favor of LookupResources
+	//       the only issue here is that the only caller of this is not doing anything with the ctype
+	//       so we might have to change the signature of LookupResources to skip the ctype, which might be reasonable
+	if accRef == nil {
+		return data, trackerdb.ErrNotFound
+	}
+	xref := accRef.(accountRef)
+
+	value, closer, err := r.kvr.Get(resourceKey(xref.addr, aidx))
+	if err != nil {
+		return
+	}
+	defer closer.Close()
+
+	return value, nil
+}
+
+func (r *accountsReader) TotalResources(ctx context.Context) (total uint64, err error) {
+	// TODO: catchpoint
+	return
+}
+
+func (r *accountsReader) TotalAccounts(ctx context.Context) (total uint64, err error) {
+	// TODO: catchpoint
+	return
+}
+
+func (r *accountsReader) TotalKVs(ctx context.Context) (total uint64, err error) {
+	// TODO: catchpoint
+	return
+}
+
+// TODO: this replicates some functionality from LookupOnlineHistory, implemented for onlineAccountsReader
+func (r *accountsReader) LookupOnlineAccountDataByAddress(addr basics.Address) (ref trackerdb.OnlineAccountRef, data []byte, err error) {
+	low := onlineAccountOnlyPartialKey(addr)
+	high := onlineAccountOnlyPartialKey(addr)
+	high[len(high)-1]++
```

Comment:
nit: maybe have a `exclusive=true` flag or something. Plus, some methods use `+ "."`, others `high[len(high)-1]++`, we need some consistency here